saleor/graphql/product/mutations/products.py

```diff
@@ -1484,6 +1501,12 @@ def perform_mutation(cls, _root, info, image_id, variant_id):
                     }
                 )
         variant = ChannelContext(node=variant, channel_slug=None)
+
+        with transaction.atomic():
```

Comment:
See my comment above comment according to transaction.
Why do we call call variant_delete for mutation which assigns images?

Reply:
Typo

Reply:
You're creating transaction only to call  on_commit inside. See my previous comment.